crash in snapshot.tree

I'm probably invoking the command incorrectly, but at least it shouldnt crash?

```
.\govc.exe snapshot.tree -vm "mar*" -u "usr:pwd@host"
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x514a53]

goroutine 1 [running]:
panic(0xb41740, 0xc042004080)
        C:/Users/vm/AppData/Local/scoop/apps/go/1.7.1/src/runtime/panic.go:500 +0x1af
github.com/vmware/govmomi/govc/vm/snapshot.(*tree).Run(0xc0420c3500, 0x12151a0, 0xc042042220, 0xc042038a20, 0x0, 0x0)
        C:/Users/vm/dev/go/src/github.com/vmware/govmomi/govc/vm/snapshot/tree.go:122 +0x243
github.com/vmware/govmomi/govc/cli.Run(0xc042038190, 0x5, 0x5, 0xc04201a000)
        C:/Users/vm/dev/go/src/github.com/vmware/govmomi/govc/cli/command.go:149 +0x58d
main.main()
        C:/Users/vm/dev/go/src/github.com/vmware/govmomi/govc/main.go:72 +0x6d
```

--- a/govc/test/vm.bats
+++ b/govc/test/vm.bats
@@ -483,6 +483,10 @@ load test_helper
   vm=$(new_ttylinux_vm)
   id=$(new_id)
 
+  # No snapshots == no output
+  run govc snapshot.tree -vm "$vm"
+  assert_success ""
+
   run govc snapshot.remove -vm "$vm" '*'
   assert_success
 

--- a/govc/vm/snapshot/tree.go
+++ b/govc/vm/snapshot/tree.go
@@ -54,6 +54,16 @@ func (cmd *tree) Register(ctx context.Context, f *flag.FlagSet) {
 	f.BoolVar(&cmd.id, "i", false, "Print the snapshot id")
 }
 
+func (cmd *tree) Description() string {
+	return `List VM snapshots in a tree-like format.
+
+The command will exit 0 with no output if VM does not have any snapshots.
+
+Examples:
+  govc snapshot.tree -vm my-vm
+  govc snapshot.tree -vm my-vm -D -i`
+}
+
 func (cmd *tree) Process(ctx context.Context) error {
 	if err := cmd.VirtualMachineFlag.Process(ctx); err != nil {
 		return err
@@ -117,6 +127,10 @@ func (cmd *tree) Run(ctx context.Context, f *flag.FlagSet) error {
 	err = vm.Properties(ctx, vm.Reference(), []string{"snapshot"}, &o)
 	if err != nil {
 		return err
+	}
+
+	if o.Snapshot == nil {
+		return nil
 	}
 
 	if cmd.current && o.Snapshot.CurrentSnapshot == nil {

